Applications/TextEditor/TextEditorWidget.cpp

```diff
@@ -33,17 +39,33 @@ TextEditorWidget::TextEditorWidget()
     });
 
     auto open_action = GAction::create("Open document", { Mod_Ctrl, Key_O }, GraphicsBitmap::load_from_file("/res/icons/16x16/open.png"), [this](const GAction&) {
-        GFilePicker picker;
-
-        if (picker.exec() == GDialog::ExecOK) {
-            m_path = picker.selected_file().string();
+        Optional<String> open_name = GFilePicker::get_open_filepath();
+        if (open_name.has_value() && !open_name.value().is_null()) {
```

Comment:
`get_open_filepath()` should return an `Optional<String>`. Then you don't need to check `is_null()`.
The plan is for null `String` objects to go away eventually, so when we need to signal absence we should use `Optional` :)